### ISSUE TITLE
Users with LDAP authentication can log in without having to enter the password

### ISSUE THREAD

**Issue by 97carmine** [jellyfin/jellyfin#1445]
**Describe the bug**
In 10.3.3, users with LDAP authentication can log in without problems, with username and password.

In 10.3.4, users with LDAP authentication do not have the need to enter the password to log in, just the user name is enough.

The LDAP Authentication plugin does not work. showing the following error in the log.

```
[2019-06-07 19:32:01.818 +02:00] [INF] Loaded assembly "LDAP-Auth, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/LDAP Authentication/LDAP-Auth.dll"
[2019-06-07 19:32:01.818 +02:00] [ERR] Error getting exported types from "LDAP-Auth, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null"
System.TypeLoadException: Method 'ChangeEasyPassword' in type 'Jellyfin.Plugin.LDAP_Auth.LdapAuthenticationProviderPlugin' from assembly 'LDAP-Auth, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes)
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at Emby.Server.Implementations.ApplicationHost.GetTypes(IEnumerable`1 assemblies)+MoveNext()
```

When trying to install it also shows an error in the log.

```
[2019-06-07 19:24:53.902 +02:00] [INF] New plugin installed: "LDAP Authentication" "2.0" Release
[2019-06-07 19:24:53.909 +02:00] [FTL] Unhandled Exception
System.TypeLoadException: Method 'ChangeEasyPassword' in type 'Jellyfin.Plugin.LDAP_Auth.LdapAuthenticationProviderPlugin' from assembly 'LDAP-Auth, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes)
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Emby.Server.Implementations.ApplicationHost.PluginInstalled(Object sender, GenericEventArgs`1 args)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**System (please complete the following information):**
 - OS: Debian
 - Jellyfin Version: 10.3.4
 - Reverse proxy: no

**Comment by joshuaboniface** [comment]
This has been confirmed. We've pulled the 10.3.4 release until this can be more thoroughly investigate.

I was unable to get the errors to show, however what appears to happen is that all LDAP-only users have had their Authentication Provider reset to be "Default", which means they effectively have no passwords. This persists even after downgrading to 10.3.3 - you should modify the users to set their authentication provider back to LDAP immediately.

**Comment by joshuaboniface** [comment]
I've done some more testing and understand what's going on here. There's actually two bugs in play.

1. The LDAP auth plugin is broken in 10.3.4, throwing the error mentioned due to a missing function required by #1406. This causes the LDAP plugin to fail to load, and can be addressed in the plugin itself. However this is causing the following issue to trigger.

2. The more underlying bug is with how users are authenticated. This bug affects *every* version for all LDAP-only users *if the LDAP plugin fails to function or is removed* for any reason. It would likely also affect other authentication providers if we would have had them.

In short, if a user us configured with `<AuthenticationProviderId>Jellyfin.Plugin.LDAP_Auth.LdapAuthenticationProviderPlugin</AuthenticationProviderId>` in their `policy.xml`, and the specified authentication provider is not available when the user attempts to log in, Jellyfin resets this to `<AuthenticationProviderId>Emby.Server.Implementations.Library.DefaultAuthenticationProvider</AuthenticationProviderId>`. And since the user now has no local password, since one is not set on user creation in the LDAP authentication plugin, the user is now free to log in without a password. This is the reason the issue is not fixed via a revert (or a correction of the LDAP plugin) after the bug is triggered.

We are exploring long-term solutions, but as a quick security workaround one can revert back to 10.3.3 (if needed), ensure the LDAP plugin is working as expected, then for each LDAP-only user:
    a) Temporarily change the user to the Default Authentication provider via the user settings page.
    b) Set a password for the user of some kind.
    c) Change the user back to the LDAP Authentication provider.

This will secure the user even if the authentication provider fails by ensuring that the Jellyfin user record has at least some sort of password.

**Comment by joshuaboniface** [comment]
This has been fixed permanently with a fail-closed default, and the updated LDAP auth plugin will be published very shortly along with the 10.3.5 release.
